lib/utils.js

```diff
@@ -97,6 +97,8 @@ var Utils = module.exports = {
     return lodash.cloneDeep(obj, function (elem) {
       // Unfortunately, lodash.cloneDeep doesn't preserve Buffer.isBuffer, which we have to rely on for binary data
       if (Buffer.isBuffer(elem)) { return elem; }
+      // Preserve data-types like `fn` across clones
+      if (elem instanceof Utils.fn) { return new Utils.fn(elem.fn, elem.args); }
```

Comment:
A `clone` method would be super, deep clone can then look for the presence of that on elem (generally) and call it if need be.